weave/trace_server/external_to_internal_trace_server_adapter.py

```diff
@@ -465,215 +467,269 @@ def evaluation_status(
         req.project_id = self._idc.ext_to_int_project_id(req.project_id)
         return self._ref_apply(self._internal_trace_server.evaluation_status, req)
 
-    # === V2 APIs ===
+    @cached_property
+    def object_interface(self) -> oi.ObjectInterface:
```

Comment:
another smell - you have to have a 2 way binding between these objects (.object_interface and ._server) ...